poseidon2/benches/poseidon2.rs

```diff
@@ -1,50 +1,63 @@
     let mut rng = thread_rng();
-    // poseidon2_p64::<BabyBear, Poseidon2ExternalMatrixGeneral, DiffusionMatrixBabyBear, 16, 7>(c);
-    // poseidon2_p64::<BabyBear, Poseidon2ExternalMatrixGeneral, DiffusionMatrixBabyBear, 24, 7>(c);
 
-    // poseidon2_p64::<KoalaBear, Poseidon2ExternalMatrixGeneral, DiffusionMatrixKoalaBear, 16, 3>(c);
-    // poseidon2_p64::<KoalaBear, Poseidon2ExternalMatrixGeneral, DiffusionMatrixKoalaBear, 24, 3>(c);
+    let poseidon2_bb_16 = Poseidon2BabyBear::<16>::new_from_rng_128(&mut rng);
+    poseidon2::<BabyBear, Poseidon2BabyBear<16>, 16>(c, poseidon2_bb_16);
+    let poseidon2_bb_24 = Poseidon2BabyBear::<24>::new_from_rng_128(&mut rng);
+    poseidon2::<BabyBear, Poseidon2BabyBear<24>, 24>(c, poseidon2_bb_24);
+
+    let poseidon2_kb_16 = Poseidon2KoalaBear::<16>::new_from_rng_128(&mut rng);
+    poseidon2::<KoalaBear, Poseidon2KoalaBear<16>, 16>(c, poseidon2_kb_16);
+    let poseidon2_kb_24 = Poseidon2KoalaBear::<24>::new_from_rng_128(&mut rng);
+    poseidon2::<KoalaBear, Poseidon2KoalaBear<24>, 24>(c, poseidon2_kb_24);
 
     let poseidon2_m31_16 = Poseidon2Mersenne31::<16>::new_from_rng_128(&mut rng);
-    poseidon2::<Mersenne31, Poseidon2Mersenne31<16>, 16, 5>(c, poseidon2_m31_16);
+    poseidon2::<Mersenne31, Poseidon2Mersenne31<16>, 16>(c, poseidon2_m31_16);
     let poseidon2_m31_24 = Poseidon2Mersenne31::<24>::new_from_rng_128(&mut rng);
-    poseidon2::<Mersenne31, Poseidon2Mersenne31<24>, 24, 5>(c, poseidon2_m31_24);
+    poseidon2::<Mersenne31, Poseidon2Mersenne31<24>, 24>(c, poseidon2_m31_24);
 
     let poseidon2_gold_8 = Poseidon2Goldilocks::<8>::new_from_rng_128(&mut rng);
-    poseidon2::<Goldilocks, Poseidon2Goldilocks<8>, 8, 5>(c, poseidon2_gold_8);
+    poseidon2::<Goldilocks, Poseidon2Goldilocks<8>, 8>(c, poseidon2_gold_8);
     let poseidon2_gold_12 = Poseidon2Goldilocks::<12>::new_from_rng_128(&mut rng);
-    poseidon2::<Goldilocks, Poseidon2Goldilocks<12>, 12, 5>(c, poseidon2_gold_12);
+    poseidon2::<Goldilocks, Poseidon2Goldilocks<12>, 12>(c, poseidon2_gold_12);
     let poseidon2_gold_16 = Poseidon2Goldilocks::<16>::new_from_rng_128(&mut rng);
-    poseidon2::<Goldilocks, Poseidon2Goldilocks<16>, 16, 5>(c, poseidon2_gold_16);
+    poseidon2::<Goldilocks, Poseidon2Goldilocks<16>, 16>(c, poseidon2_gold_16);
+
+    // We hard code the round numbers for Bn254Fr.
+    let poseidon2_bn254 = Poseidon2Bn254::<3>::new_from_rng(8, 22, &mut rng);
+    poseidon2::<Bn254Fr, Poseidon2Bn254<3>, 3>(c, poseidon2_bn254);
+}
 
-    // poseidon2::<Bn254Fr, Poseidon2ExternalMatrixGeneral, DiffusionMatrixBN254, 3, 5>(c, 8, 22);
+// TODO: REMOVE BEFORE MERGING WITH MAIN.
+// CAN IMPORT THIS FROM UTILS ONCE WE UPDATE TO THE LATEST VERSION OF MAIN
+fn pretty_name<T>() -> String {
+    let name = type_name::<T>();
+    let mut result = String::new();
+    for qual in name.split_inclusive(&['<', '>', ',']) {
+        result.push_str(qual.split("::").last().unwrap());
+    }
+    result
```

Comment:
Yeah I think it should be fine. I was planning to wait to merge this into angus/vectorized_poseidon2_production just to avoid having to incorporate main twice (Both into angus/vectorized_poseidon2_production and this branch).